corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js

```diff
@@ -626,12 +626,19 @@ function GeoPointEntry(question, options) {
             self.map.setZoom(self.DEFAULT.anszoom);
         }
         google.maps.event.addListener(self.map, "center_changed", self.updateCenter.bind(self));
-    }
+    };
+
     self.afterRender = function() {
-        if (typeof google === "undefined") {
-            $.getScript(self.apiKey + '&callback=gMapsCallback');
+        if (typeof google === "undefined" && !window.gMapsRequested) {
+            // First entry to attempt to load google
+            window.gMapsRequested = true;
+            $.getScript(self.apiKey, self.gMapsCallback);
+        } else if (typeof google === "undefined" && window.gMapsRequested) {
+            // Waiting for gmaps to load, recursively call afterRender
+            setTimeout(self.afterRender, 400);
```

Comment:
this solution is meh, but any other solution i think is fairly non-trivial unless we want to always load the google api script